### DISCUSSION TITLE
Edirol SD-80 panel, but the creator knows nothing about UI design or programming

### DISCUSSION THREAD

**Discussion by IceDragonDee** [RomanKubiak/ctrlr#580] [Show and tell]
As the title mentions, I'm not a UI designer or a programmer. What I am though, is sick of Roland's official editor for this synth.

I hardly know what I'm doing with Ctrlr and in fact, there are many things about this software that confuse me or I outright dislike even, but seeing as no one else can be bothered to make an actually good editor for this synth, I have no choice but to do it myself... But I don't know what I'm doing at all.

I don't know what I'm doing at all, but I've been recommended to not use the uiTab component and instead do everything with layers and LUA. I'm incredibly confused by all of this, but I don't know what other options I have.

I've been looking at pictures of the JV-XP panel for reference, but frankly, I feel like I'm way out of my depth.
Can't seem to attach the picture of what little I have though, but so far, it's just 16 mute buttons, volume sliders, pan knobs and patch recall/get buttons. None of these buttons do anything. I think I should get a UI designed before setting it up to work.
I probably should try making something simpler, but everything else I have either isn't simpler or doesn't need a software editor.

**Comment by Ted-juh** [comment]
Easy to say but start somewhere. You need midi or system-exclusive messages to do anything with those sliders and knobs. Starting at page 49 of the owners' manual, there is some talk about getting dumps and how to construct some system-exclusive messages. There is also a midi implementation chart. Although I couldn't find a full system-exclusives chart, the Roland editor displays the system-exclusive messages. 

Start small, slam one button in Ctrlr, and try to make it switch sound generator modes. (page 53), Then add a slider. Get the hang of it, see what all those options do in the Ctrlr sidebar. Fiddle around. Read about and practice some Lua. Get 1000 errors without a clue why. But the minute a button or a slider works, you're sold. 

But to be honest, yeah, you could have chosen an easier editor. The amount of parameters on that device is astonishing. But in the end, it depends on what you want to control. But, in my opinion, layout or UI is the last step to bother about. First, get your programming skills into 1st gear. Because if you want layers, then yes, some programming skills come in handy. Ever taken a bulk dump and wondered what all those hex values stand for? Again, you need Lua to do anything useful with it. So.. start somewhere but start small.

**Reply by IceDragonDee** [reply]
All I know is that the SD-90 MIDI implementation chart is available, which is very similar except for that it has no ability to save.

I've been speaking with friends for several days since before making this discussion and I have recently found that I'm not in the right headspace to continue. Stress is too high and my mental state isn't right. I'll try coming back to this after I've moved out.

I was hoping to use the JV-XP panel as a jumping off point as that synth family is similar to the XV and StudioCanvas family, but with no .panel file to pull from, I'm not capable to figure this out.
